javascript/javascript3/week2/visual.js

```diff
@@ -0,0 +1,63 @@
+//target move one by one
+async function translateOneByOne() {
+    await new Promise((resolve) => {
+      resolve(
+        moveElement(document.querySelector("ul.marks li:nth-child(1)"), {
+          x: 20,
+          y: 300,
+        })
+      );
+      console.log("first");
+    });
+    await new Promise((resolve) => {
+      resolve(
+        moveElement(document.querySelector("ul.marks li:nth-child(2)"), {
+          x: 400,
+          y: 300,
+        })
+      );
+      console.log("first");
+    });
+    await new Promise((resolve) => {
+      resolve(
+        moveElement(document.querySelector("ul.marks li:nth-child(3)"), {
+          x: 400,
+          y: 20,
+        })
+      );
+      console.log("first");
+    });
+  }
+  
+  translateOneByOne();
+//target move all at once 
+//comment this code for check target move one by one
+  function translateAllAtOnce() {
+    const p1 = new Promise((resolve) => {
+      resolve(
+        moveElement(document.querySelector("ul.marks li:nth-child(1)"), {
+          x: 20,
+          y: 300,
+        })
+      );
+    });
+    const p2 = new Promise((resolve) => {
+      resolve(
+        moveElement(document.querySelector("ul.marks li:nth-child(2)"), {
+          x: 400,
+          y: 300,
+        })
+      );
+    });
+    const p3 = new Promise((resolve) => {
+      resolve(
+        moveElement(document.querySelector("ul.marks li:nth-child(3)"), {
+          x: 400,
+          y: 20,
+        })
+      );
+    });
+    Promise.all([p1, p2, p3]);
+  }
+  
+  translateAllAtOnce();
```

Comment:
No need to wrap `moveElement` with a promise. It is a function when you call it returns a promise. 
Take a lock at this 
```js
function translateOneByOne() {
  moveElement(red, target.red)
    .then(() => {
      console.log('red moved');
      return moveElement(blue, target.blue);
    })
    .then(() => {
      console.log('blue moved');
      return moveElement(green, target.green);
    })
    .then(() => {
      console.log('green moved');
    });
}
```